### ISSUE TITLE
Compile of packages/os400/chkstrings.c fails

### ISSUE THREAD

**Issue by johnsherrill** [curl/curl#9911]
Compile fails because "curl_socklen_t" is not defined.  Include "setup-os400.h" defines this structure
extern int Curl_getnameinfo_a(const struct sockaddr *sa,                                          |    623     6
    55       |                              curl_socklen_t salen,                                               |    624     6
 ===========> ..............................a...................................................................
 *=SEVERE==========> a - CZM0046  Syntax error.
    56       |                              char *nodename, curl_socklen_t nodenamelen,                         |    625     6
    57       |                              char *servname, curl_socklen_t servnamelen,                         |    626     6
    58       |                              int flags); 

curl_socklen_t is defined in "system.h" 
adding #include <system.h> to setup-os400.h fixes the problem.

**Comment by bagder** [comment]
But setup-os400.h does not use `curl_socklen_t` ?

https://github.com/curl/curl/blob/f4912407eb37ab12a09aa124dd5ce5be3ff03932/lib/setup-os400.h#L55-L56

**Comment by johnsherrill** [comment]
You are right there is an updated version of setup-os400.h and os400sys.c that fixes the issue. Thanks.
